http-clients/apache5-client/src/main/java/software/amazon/awssdk/http/apache5/internal/RepeatableInputStreamRequestEntity.java

```diff
@@ -149,8 +121,8 @@ public boolean isRepeatable() {
     @Override
     public void writeTo(OutputStream output) throws IOException {
         try {
-            if (!firstAttempt && isRepeatable()) {
-                content.reset();
+            if (!firstAttempt && isRepeatableStream()) {
+                getContent().reset();
```

Comment:
Is there any reason we are calling getContent() instead of saving it as a member variable? `ContentStreamProvider#newStream` could be very expensive for some custom implementations.

Reply:
done